test/prototype/test_low_bit_optim.py

```diff
@@ -376,6 +397,40 @@ def _test_fsdp2(self, optim_cls):
 
         self.assertEqual(base_exp_avg.dequantize(), full_fsdp_exp_avg.dequantize())
 
+        # test for compatibility with dcp.save() and .load()
+        checkpoint_id = f"_fsdp_low_bit_optim_{optim_cls.__name__}"
+        if Path(checkpoint_id).exists():
+            shutil.rmtree(checkpoint_id)
+        dcp.save(fsdp_optim.state_dict(), checkpoint_id=checkpoint_id)
+
+        # normally we would want to use dcp.state_dict.get_optimizer_state_dict() to initialize optim states.
+        # however, currently it does not respect tensor-ness of LR pytorch/pytorch#139575.
+        # therefore, we have to manually initialize optim state here.
+        resumed_fsdp_optim = optim_cls(fsdp_model.parameters(), lr=1e-2)
+        for p in fsdp_model.parameters():
+            p.grad = torch.zeros_like(p)
+
+        # this will change model weights due to weight decay, but since we don't use the model anymore, it's fine.
+        resumed_fsdp_optim.step()
+
+        # NOTE: should dcp.load() have a flag for weights_only=False?
+        subclasses = (OptimState4bit, OptimState8bit, OptimStateFp8)
+        with torch.serialization.safe_globals(subclasses):
+            dcp.load(resumed_fsdp_optim.state_dict(), checkpoint_id=checkpoint_id)
```

Comment:
@msaroufim Is it ok to add these to `safe_globals` permanently when torchao is imported? i.e. run `torch.serialization.add_safe_globals(OptimState8bit)` in `subclass_8bit.py` and so on.

Reply:
should be fine - @mikaylagawarecki are there some tradeoffs we should be wary of when using safe_globals?

Reply:
Marked as resolved thanks to #1228

Reply:
`safe_globals` will add those subclasses to the list of classes/functions that are allowed to be called with arguments during unpickling for `weights_only`

Adding to safe_globals in `torchao` has been the approach taken for the rest of the torchao dtypes